modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8323,7 +8323,7 @@ public String getName() {
         void adjust(int change) {
             count += change;
 
-            if (count == 1) {
+            if (change > 0) {
                 set(true);
             } else if (count == 0) {
                 set(false);
```

Comment:
This looks a bit odd due to the use of two different variables, but it works (`count > 0` would also have worked).  I personally would favor something without any if statements as it leaves less room for misinterpretation in this case:

     count += change;

     set(count > 0);

Reply:
set(count > 0);
looks so much better

Reply:
Agreed.